engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/UserTaskFormTest.java

```diff
@@ -27,46 +34,112 @@ public class UserTaskFormTest {
+        .extracting(Record::getKey, Record::getIntent)
+        .describedAs("form not found incident is resolved and no new incident is created")
+        .containsExactly(
+            tuple(incidentCreated.getKey(), IncidentIntent.CREATED),
+            tuple(incidentCreated.getKey(), IncidentIntent.RESOLVED));
+
+    assertUserTaskActivation(processInstanceKey);
+  }
+
+  private void deployProcess(final String formId) {
     final BpmnModelInstance processWithFormId =
         Bpmn.createExecutableProcess(PROCESS_ID)
             .startEvent()
             .userTask("task")
-            .zeebeFormId("form-id")
+            .zeebeFormId(formId)
             .endEvent()
             .done();
     // given
     ENGINE.deployment().withXmlResource(processWithFormId).deploy();
+  }
 
-    // when
-    ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+  private void deployForm() {
+    final var deploymentEvent = ENGINE.deployment().withJsonClasspathResource(TEST_FORM).deploy();
 
-    // then
-    assertIncident();
+    Assertions.assertThat(deploymentEvent)
+        .hasIntent(DeploymentIntent.CREATED)
+        .hasValueType(ValueType.DEPLOYMENT)
+        .hasRecordType(RecordType.EVENT);
   }
 
-  private void assertIncident() {
-    assertThat(RecordingExporter.incidentRecords().getFirst().getValue())
+  private Record<IncidentRecordValue> assertFormIncident(
+      final long processInstanceKey, final String formId) {
+    final var incidentCreated =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    assertThat(incidentCreated.getValue())
         .extracting(r -> tuple(r.getErrorType(), r.getErrorMessage()))
         .isEqualTo(
             tuple(
                 ErrorType.FORM_NOT_FOUND,
-                "Expected to find a form with id 'form-id', but no form with this id is found, at least a form with this id should be available. To resolve the Incident please deploy a form with the same id"));
+                String.format(
+                    "Expected to find a form with id '%s', but no form with this id is found, at least a form with this id should be available. To resolve the Incident please deploy a form with the same id",
+                    formId)));
+
+    return incidentCreated;
   }
 
-  private FormRecord sampleFormRecord() {
-    return new FormRecord()
-        .setFormId("form-id")
-        .setVersion(1)
-        .setFormKey(1L)
-        .setResourceName("form-1.form")
-        .setChecksum(wrapString("checksum"))
-        .setResource(wrapString("form-resource"));
+  private void assertUserTaskActivation(final long processInstanceKey) {
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementType(BpmnElementType.USER_TASK)
+                .limit(3))
```

Comment:
❌ This limit needs to be `1` since we should observe only one user task activation for all tests.

Reply:
Nope, there should be 3 records. `ACTIVATE_ELEMENT` -> `ELEMENT_ACTIVATING` -> `ELEMENT_ACTIVATED`.

Reply:
Ahh alright, we did not assert the `ELEMENT_ACTIVATED` only. 👍